testsuite/MDAnalysisTests/coordinates/test_trc.py

```diff
@@ -0,0 +1,189 @@
+        assert len(TRC_U.trajectory) == 6
+        assert (TRC_U.trajectory.n_frames) == 6
+
+    def test_trc_n_atoms(self, TRC_U):
+        assert (TRC_U.trajectory.n_atoms) == 73
+
+    def test_trc_frame(self, TRC_U):
+        assert TRC_U.trajectory[0].frame == 0
+        assert TRC_U.trajectory[4].frame == 4
+
+    def test_trc_time(self, TRC_U):
+        assert TRC_U.trajectory[0].time == 0
+        assert TRC_U.trajectory[4].time == 80
+
+    def test_trc_dt(self, TRC_U):
+        time_array = np.array([ts.time for ts in TRC_U.trajectory])
+        assert_equal(time_array, np.arange(6)*20)
+
+        dt_array = np.diff(time_array)
+        assert_equal(dt_array, np.full(5, 20))
+
+    def test_trc_data_step(self, TRC_U):
+        assert TRC_U.trajectory[0].data['step'] == 0
+        assert TRC_U.trajectory[4].data['step'] == 10000
+
+    def test_periodic(self, TRC_U):
+        assert_equal(TRC_U.trajectory.periodic, False)
+
+    def test_rewind(self, TRC_U):
+        TRC_U.trajectory[0]
+        trc = TRC_U.trajectory
+        trc.next()
+        trc.next()
+        trc.next()
+        trc.next()
+        assert_equal(trc.ts.frame, 4,
+                     "trajectory.next() did not forward to frameindex 4")
+        trc.rewind()
+        assert_equal(trc.ts.frame, 0, "trajectory.rewind() failed "
+                                      "to rewind to first frame")
+
+        assert np.any(TRC_U.atoms.positions != 0), "The atom positions " \
+                                                   "are not populated"
+
+    def test_random_access(self, TRC_U):
+        TRC_U.trajectory[0]
+        pos0 = TRC_U.atoms[0].position
+        TRC_U.trajectory.next()
+        TRC_U.trajectory.next()
+        pos2 = TRC_U.atoms[0].position
+
+        TRC_U.trajectory[0]
+
+        assert_equal(TRC_U.atoms[0].position, pos0)
+
+        TRC_U.trajectory[2]
+
+        assert_equal(TRC_U.atoms[0].position, pos2)
+
+    def test_read_frame_reopens(self, TRC_U):
+        TRC_U.trajectory._reopen()
+        TRC_U.trajectory[4]
+        assert TRC_U.trajectory.ts.frame == 4
+
+
+class TestTRCReaderSolvatedBox:
+    @pytest.fixture(scope='class')
+    def TRC_U(self):
+        return mda.Universe(TRC_PDB_SOLV, TRC_TRAJ_SOLV)
+
+    def test_trc_n_atoms(self, TRC_U):
+        assert (TRC_U.trajectory.n_atoms) == 2797
+
+    def test_periodic(self, TRC_U):
+        assert_equal(TRC_U.trajectory.periodic, True)
+
+    def test_trc_dimensions(self, TRC_U):
+        ts = TRC_U.trajectory[1]
+        assert_allclose(
+            ts.dimensions,
+            [30.54416298, 30.54416298, 30.54416298, 90., 90., 90.]
+        )
+
+    def test_open_twice(self, TRC_U):
+        TRC_U.trajectory._reopen()
+        with pytest.raises(IOError):
+            TRC_U.trajectory.open_trajectory()
+
+
+class TestTRCReaderClusterTrajectory:
+    @pytest.fixture(scope='class')
+    def TRC_U(self):
+        with pytest.warns(UserWarning) as record_of_warnings:
```

Comment:
_maybe_ you can simplify with [pytest.warns(UserWarning, match="...")](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.warns) although I don't know how this works with two messages

Reply:
I gave some thought on that when I programmed that part of the code. As I remember, my solution was the most straight forward to check two messages. As I understood it, the `match="..."` only works with one warning. If anyone of you has a specific idea on how to make a nicer check for two warnings, i would gladly do it so. Otherwise, I want to ask to keep it like it is.